### PR TITLE
feat: environment command

### LINKED ISSUE #10583: [mesheryctl] New command exp environment
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

### Desired Behavior
Develop an experimental command to manage environments within the system. Refer to the [Environment documentation](https://docs.meshery.io/concepts/logical/environments) for more details about environments.

Tasks to be Completed:
- [ ] Implement command to list existing environments
- [ ] Implement command to create new environments
- [ ] Implement command to delete existing environments
- [ ] Implement command to view details of a specific environment
- [ ] Develop comprehensive test cases to validate the functionality of the environment command

---

### Contributor Resources

- ⌨️ [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](http://discuss.meshery.io) and [Layer5 Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by RipulHandoo**
**Notes for Reviewers**

This PR fixes #10583 
- [x] Implement command to list environment
- [x]  Implement command to create new environment
- [x] Implement command to view environments
- [x] Implement command to delete an environment
- [ ]  Develop test cases to ensure command functionality

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
merge conflicts...

**Review by MUzairS15 — CHANGES_REQUESTED** [review]
@RipulHandoo Go through previous PR reviews, this PR also suffers from the same issues. Logs, error, returning err vs logging

**Comment by RipulHandoo** [comment]
> @RipulHandoo Go through previous PR reviews, this PR also suffers from the same issues. Logs, error, returning err vs logging

Done. Reviews answered.

**Comment by leecalcote** [comment]
@RipulHandoo merge conflicts...

**Comment by RipulHandoo** [comment]
> @RipulHandoo merge conflicts...

Resolved.

File: mesheryctl/internal/cli/root/environments/delete.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/config"
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/system"
+	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var DeleteEnvironmentCmd = &cobra.Command{
```
Comment (MUzairS15) [inline mesheryctl/internal/cli/root/environments/delete.go:30]:
```suggestion
var deleteEnvironmentCmd = &cobra.Command{
```

File: mesheryctl/internal/cli/root/environments/create.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/system"
+	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
+	"github.com/layer5io/meshery/server/models/environments"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var CreateEnvironmentCmd = &cobra.Command{
```
Comment (MUzairS15) [inline mesheryctl/internal/cli/root/environments/create.go:33]:
```suggestion
var createEnvironmentCmd = &cobra.Command{
```

File: mesheryctl/internal/cli/root/environments/environment.go [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+	name          string
+	description   string
+	orgID         string
+	outFormatFlag string
+	saveFlag      bool
+
+	maxRowsPerPage    = 25
+	whiteBoardPrinter = color.New(color.FgHiBlack, color.BgWhite, color.Bold)
+	// availableSubcommands = []*cobra.Command{listEnvironmentCmd, CreateEnvironmentCmd, DeleteEnvironmentCmd, viewEnvironmentCmd}
+	availableSubcommands = []*cobra.Command{listEnvironmentCmd, CreateEnvironmentCmd, DeleteEnvironmentCmd}
```
Comment (MUzairS15) [inline mesheryctl/internal/cli/root/environments/environment.go:42]:
```suggestion
	availableSubcommands = []*cobra.Command{listEnvironmentCmd, createEnvironmentCmd, deleteEnvironmentCmd}
```

File: mesheryctl/internal/cli/root/environments/environment.go [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+		}
+		return nil
+	},
+}
+
+func init() {
+	listEnvironmentCmd.Flags().StringVarP(&orgID, "orgId", "o", "", "Organization ID")
+	viewEnvironmentCmd.Flags().StringVarP(&outFormatFlag, "output-format", "o", "yaml", "(optional) format to display in [json|yaml]")
+	viewEnvironmentCmd.Flags().BoolVarP(&saveFlag, "save", "s", false, "(optional) save output as a JSON/YAML file")
+	CreateEnvironmentCmd.Flags().StringVarP(&orgID, "orgId", "o", "", "Organization ID")
```
Comment (MUzairS15) [inline mesheryctl/internal/cli/root/environments/environment.go:107]:
```suggestion
	createEnvironmentCmd.Flags().StringVarP(&orgID, "orgId", "o", "", "Organization ID")
```

File: mesheryctl/internal/cli/root/environments/environment.go [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+	CreateEnvironmentCmd.Flags().StringVarP(&name, "name", "n", "", "Name of the environment")
```
Comment (MUzairS15) [inline mesheryctl/internal/cli/root/environments/environment.go:108]:
```suggestion
	createEnvironmentCmd.Flags().StringVarP(&name, "name", "n", "", "Name of the environment")
```

File: mesheryctl/internal/cli/root/environments/environment.go [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+	CreateEnvironmentCmd.Flags().StringVarP(&description, "description", "d", "", "Description of the environment")
```
Comment (MUzairS15) [inline mesheryctl/internal/cli/root/environments/environment.go:109]:
```suggestion
	createEnvironmentCmd.Flags().StringVarP(&description, "description", "d", "", "Description of the environment")
```

File: mesheryctl/internal/cli/root/environments/view.go [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+		if outFormatFlag != "json" && outFormatFlag != "yaml" {
+			return errors.New("output-format choice is invalid or not provided, use [json|yaml]")
+		}
+		// Get the home directory of the user to save the output file
+		homeDir, _ := os.UserHomeDir()
+		componentString := strings.ReplaceAll(fmt.Sprintf("%v", selectedEnvironment.Name), " ", "_")
+
+		if outFormatFlag == "yaml" {
+			if output, err = yaml.Marshal(selectedEnvironment); err != nil {
+				return errors.Wrap(err, "failed to format output in YAML")
```
Comment (MUzairS15) [inline mesheryctl/internal/cli/root/environments/view.go:137]: Please update to meshkit error.

File: mesheryctl/internal/cli/root/environments/view.go [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+			}
+			if saveFlag {
+				utils.Log.Info("Saving output as YAML file")
+				err = os.WriteFile(homeDir+"/.meshery/component_"+componentString+".yaml", output, 0666)
+				if err != nil {
+					return errors.Wrap(err, "failed to save output as YAML file")
```
Comment (MUzairS15) [inline mesheryctl/internal/cli/root/environments/view.go:143]: same

File: mesheryctl/internal/cli/root/environments/view.go [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+		outFormatFlag = strings.ToLower(outFormatFlag)
+
+		if outFormatFlag != "json" && outFormatFlag != "yaml" {
+			return errors.New("output-format choice is invalid or not provided, use [json|yaml]")
+		}
+		// Get the home directory of the user to save the output file
+		homeDir, _ := os.UserHomeDir()
+		componentString := strings.ReplaceAll(fmt.Sprintf("%v", selectedEnvironment.Name), " ", "_")
+
+		if outFormatFlag == "yaml" {
```
Comment (MUzairS15) [inline mesheryctl/internal/cli/root/environments/view.go:135]: yaml | yml?

File: mesheryctl/internal/cli/root/environments/view.go [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+				err = os.WriteFile(homeDir+"/.meshery/component_"+componentString+".json", output, 0666)
+				if err != nil {
+					return errors.Wrap(err, "failed to save output as JSON file")
+				}
+				utils.Log.Info("Output saved as JSON file in ~/.meshery/component_" + componentString + ".json")
+				return nil
+			}
+			return components.OutputJson(selectedEnvironment)
+		} else {
+			return errors.New("output-format choice invalid, use [json|yaml]")
```
Comment (MUzairS15) [inline mesheryctl/internal/cli/root/environments/view.go:165]: same

**Review by MUzairS15 — DISMISSED** [review]
Please fix formatting (run gofmt)

File: mesheryctl/internal/cli/root/environments/view.go (hunk 10 above)
Comment (RipulHandoo) [reply mesheryctl/internal/cli/root/environments/view.go:165]: Done with requested changes.

**Review by MUzairS15 — APPROVED** [review]

**Pull request merged**